openai/azure-openai/runtime/src/main/java/io/quarkiverse/langchain4j/azure/openai/runtime/config/LangChain4jAzureOpenAiConfig.java

```diff
@@ -77,8 +81,7 @@ interface AzureAiConfig {
         /**
          * Azure OpenAI API key
          */
-        @WithDefault("dummy") // TODO: this should be optional but Smallrye Config doesn't like it..
-        String apiKey();
+        Optional<String> apiKey();
```

Comment:
Have you tested this with multiple (named) models? That is where it stared failing for me a while back.

Reply:
> Have you tested this with multiple (named) models? That is where it stared failing for me a while back.

I have just tested it with the following configuration.

```
#quarkus.langchain4j.azure-openai.api-key=token1
quarkus.langchain4j.azure-openai.openai1.resource-name=<<myname>>
quarkus.langchain4j.azure-openai.openai1.deployment-name=gpt-35-turbo
quarkus.langchain4j.azure-openai.openai1.api-version=2023-07-01-preview

quarkus.langchain4j.azure-openai.openai2.azure-ad-token=token2
quarkus.langchain4j.azure-openai.openai2.resource-name=<<myname>>
quarkus.langchain4j.azure-openai.openai2.deployment-name=gpt-35-turbo
quarkus.langchain4j.azure-openai.openai2.api-version=2023-07-01-preview


quarkus.langchain4j.embedding-model.provider=dev.langchain4j.model.embedding.AllMiniLmL6V2QuantizedEmbeddingModel
quarkus.langchain4j.redis.dimension=384
rag.path=src/main/resources/catalog

quarkus.log.level=INFO
```

Here are my observations.

In the `AzureOpenAiRecorder` there are 3 configurations that are created. 
- `<default>` --no matter what you do, this is always created
- `openai1`
- `openai2`

The problem lies with the Default configuration. In the above configuraiton, since it's always created, the default configuration will result in an error inside the code when checking the validity of the configuration, thus crashing the rest of the application. This will be hidden from the user. It seems that smallrye works somewhat fine in this scenario, it's the assumptions in the code that make it unusable.

Temporary resolution: 
Put `quarkus.langchain4j.azure-openai.enable-integration=false` in application.properties of the main application, thus making the recorder not check the default configuration at all.

Proposal for permanent solution: 
Make assumptions as to how a configuration is valid. A configuration should be valid if at least one of the tokens is present, otherwise it should not be checked.

I can take a look if you want, I will just need some advice on the intended usage...

Reply:
It turns out that `<default>` was being requested because of an AiService being used

Reply:
I have also opened #477  to improve the test coverage because this case did get me wondering whether we were missing something.

Reply:
Ah, yes, it was entirely my fault, sorry about that.

Reply:
No problem :)